### DISCUSSION TITLE
有没有可能用这个机器人玩地牢游戏？

### DISCUSSION THREAD

**Discussion by godblessbb** [zhayujie/chatgpt-on-wechat#463] [Ideas]
在git上看到了这个项目：https://github.com/bupticybee/ChineseAiDungeonChatGPT
这个项目其实也是脱胎于美国的一个自动驾驶工程师的项目，该工程师使用gpt模型设计了一个文字游戏，他会给定一个故事背景、宝物和英雄名字等信息，作为整个项目的描述持续输入。用户输入一个祈使句，然后ai讲祈使句拓展出更多的故事细节并推进故事。

然后我就想到可不可以用这个机器人实现这个功能？需要开发吗？还是在角色设定那里做好修改就能达到同样的效果？求教大神！

**Comment by zhayujie** [comment]
看了下这个仓库， 你把 故事背景填在 character_desc 参数试试，比如

"character_desc": "你在树林里冒险，指不定会从哪里蹦出来一些奇怪的东西，你握紧手上的手枪，希望这次冒险能够找到一些值钱的东西，你往树林深处走去"

**Comment by lanvent** [comment]
看到蛮好玩的，用插件写了个，可以试一下。https://github.com/lanvent/chatgpt-on-wechat/tree/dev
开启Dungeon插件后（初始时自动开启），输入以下命令触发
“$开始冒险 <背景故事>” - 以<背景故事>开始一个地牢游戏，不填写会使用默认背景故事。之后聊天中你的所有消息会帮助ai完善这个故事。
"$停止冒险" - 停止一个地牢游戏，回归正常的ai。

**Comment by lichengzhe** [comment]
之前想过/role选择各种角色，替换默认character_desc t，可参考人设大全https://prompts.chat 集成过来定期更新，作家，佛祖，心理咨询师，动漫角色

**Comment by lichengzhe** [comment]
role 佛祖
role 猫娘
role 程序GPT
role new "律师" “你是一个法务顾问，你。。。"
role save 
role list
role update

role的持久化，社区化，system prompt的价值和挖掘，虽然官方文档提到目前system注意力可能不足，但实测已经非常有效，并且官网也有提出后面system会更灵光，建议role对应system，如在user实现，可考虑"置顶" prompt 0方案，避免超长被挤出记忆。
<img width="612" alt="image" src="https://user-images.githubusercontent.com/11499281/226215233-4634e445-43af-4769-b5d4-e1541cdd1e6d.png">
